### ISSUE TITLE
NVDA sometimes cannot read math

### ISSUE THREAD

**Issue by CyrilleB79** [nvaccess/nvda#19813]
### Steps to reproduce

When trying to read math, NVDA sometimes cannot read it. This happens only intermittently, but quite often though.

### Actual behavior

When reaching math,, NVDA just don't report anything.

The following message is logged:
```
DEBUGWARNING - NVDAObjects.IAccessible.ia2Web.Math._get_mathMl (16:50:46.406) - MainThread (15252):
Error retrieving math. Not supported in this browser or ISimpleDOM COM proxy not registered.
Traceback (most recent call last):
  File "NVDAObjects\IAccessible\ia2Web.pyc", line 336, in _get_mathMl
  File "comtypes\_post_coinit\unknwn.pyc", line 406, in QueryInterface
  File "monkeyPatches\comtypesMonkeyPatches.pyc", line 33, in __call__
_ctypes.COMError: (-2147467262, 'Cette interface n’est pas prise en charge', (None, None, None, 0, None))
```

Restarting NVDA and moving the focus out of the browser and back in the browser (to prevent Chrome from using UIA after restart as documented in #13704) does not solve the issue.

Restarting Chrome, paying attention to close all Chrome windows, seems to solve the issue.

### Expected behavior
NVDA should read math as expected
### NVDA logs, crash dumps and other attachments
N/A
#### NVDA installed/portable/running from source
Installed
#### NVDA version
2026.1beta7
#### Windows version
Windows 11 23H2 (AMD64) build 22631.6649
#### Name and version of other software in use when reproducing the issue
Chrome 146.0.7680.80
#### Other information about your system
N/A
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors
The issue was already present in older beta (at least 2026.1beta6).
Not sure but I guess that the issue was already present on NVDA 2025.3 with MathCAT add-on and maybe with Access8Math too.
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
The tool has been run during NVDA 2026.1beta7 installation process.

**Comment by CyrilleB79** [comment]
If other people face the same issue, please comment here so that we can figure if the issue is widespread and if we can identify more precise conditions in which the issue occurs.

**Comment by RyanMcCleary** [comment]
A user on the BlindMath mailing list reported the same error a few days ago using NVDA 2025.3.3 with the MathCAT add-on (MathCAT 0.7.5).

**Comment by SaschaCowley** [comment]
@CyrilleB79 can you provide a full debug log of this happening, or at the very least dev info when it happens?

**Comment by CyrilleB79** [comment]
@SaschaCowley, here is the log:
[nvda.log](https://github.com/user-attachments/files/26207201/nvda.log)

**Comment by CyrilleB79** [comment]
Please ignore my previous message log. Here is a reproducible STR and the associated logs:
* Start NVDA
* Open a webpage containing mathml, e.g. [MathML Examples](https://support.freedomscientific.com/SurfsUp/MathML-Samples.html)
* Check that you can read math with MathCAT
* Press `NVDA+q` and choose to restart NVDA; do not use `control+alt+n` instead since it would focus the task bar.
* When NVDA has restarted, read the page. As expected due to #13704, NVDA uses UIA for Chrome. So math is not read with MathCAT (since MathCAT does not work with Chrome UIA).
* Move the focus away and back (e.g. focusing the window menu with pressing `alt`, or with `alt+tab` to another application) to trigger a focus event so that "Use UIA with Microsoft Edge and other 
Chromium based browsers when available" option = "Only when necessary" be taken into account, i.e. Chrome should finally use IAccessible.
* Try to read math

Result: Although Chrome is finally using IAccessible, math cannot be read.

[nvda.log](https://github.com/user-attachments/files/26208077/nvda.log)
[nvda-old.log](https://github.com/user-attachments/files/26208086/nvda-old.log)

Note: I have not the following in the log. But if you close all Chrome windows and reopen it, you can read math again with MathCAT.

By the way:
1. Have we logged an issue indicating that MathCAT does not work with Chrome UIA?
2. Since MathCAT does not work in all cases, e.g. Chrome UIA, should we reopen #17667?

**Comment by GeorgeKerscher** [comment]
I have not had this problem in Chrome, Edge, Thorium, and Paperback. The MathML is read. Pressing enter in the expression puts me in the MathCAT UI.

**Comment by CyrilleB79** [comment]
@GeorgeKerscher, have you tested the exact steps in my previous message?

**Comment by GeorgeKerscher** [comment]
I always have the MathML read. I can go into Word and then directly back to Chrome browser where there is MathML.

**Comment by CyrilleB79** [comment]
> I always have the MathML read. I can go into Word and then directly back to Chrome browser where there is MathML.

If you have only gone into Word and back to Crhome, that was not my exact STR...

**Comment by GeorgeKerscher** [comment]
I habe yet to have NVDA 2026.1 beta 8 not read mathml. I am going in and out of multiple applications, restarting NVDA, rebooting and starting all over and have never had a failure. I am using valid MathML expressions.

If NVDA sometimes reads the MathML and sometimes not, that is indeed a problem.  However, if there is invalid MathML, it be a problem. I have also heard about problems when MathJAX is enabled on the page. Sometimes they do not expose the MathML.

**Comment by CyrilleB79** [comment]
@GeorgeKerscher could you re-test the STR of https://github.com/nvaccess/nvda/issues/19813#issuecomment-4116684378 and provide the corresponding logs at debug level (current and previous session)? I am curious to see the difference between your setup and mine.
Also, can you confirm that you are using Chrome too?

**Comment by GeorgeKerscher** [comment]
Below is a link to my log. I   opend a file in the Chrome browser and read MathML. I opened an EPUB in Thorium and read MathML. I went to a website and read MathML. All with no problems.

[George's NVDA log](https://www.dropbox.com/scl/fi/6m08sy9bzi28isx696buw/nvda.log?rlkey=bs3yqe9pc9e2nablxidl89mrl&dl=0)

**Comment by CyrilleB79** [comment]
@GeorgeKerscher, thanks for having performed further tests.

Unfortunately, you have not enabled debug logging. So I'm not able to make a comparison between your setup and mine. Also the scope is to reproduce the steps described in https://github.com/nvaccess/nvda/issues/19813#issuecomment-4116684378; since these steps include a restart, I actually need both logs, the current one and the one of the previous NVDA session. More, I had forgotten to mention this, but you should also disable add-ons while doing these tests.

From what you say, it seems to me that you have tried to read math in a normal situation. But my point is about reading math in a very specific situation described in https://github.com/nvaccess/nvda/issues/19813#issuecomment-4116684378. In other situations, I am able to read math too.

If the instructions I have provided are not clear, just let me know.

**Comment by GeorgeKerscher** [comment]
I need some more guidance to reproduce your problem.

I should:

restart NVDA
follow your steps
Save the log
Restart NVDA with add ons disabled and debug running
follow your steps.
save the log.
Get both logs to you.

What are your steps?
If you have a link to a  page that has problematic MathML that you have encountered, that would be good to get as well.

**Comment by CyrilleB79** [comment]
@GeorgeKerscher please find below the steps to test. These are the same steps as in https://github.com/nvaccess/nvda/issues/19813#issuecomment-4116684378, but with more details.

* Press `NVDA+q`, choose "Restart with add-ons disabled and debug logging enabled" and validate to restart NVDA without add-ons and with debug log enabled.
* Open a webpage containing mathml, e.g. [MathML Examples](https://support.freedomscientific.com/SurfsUp/MathML-Samples.html)
* Using arrows, `enter` and other math navigation keys, check that you can read math with MathCAT
* Press `NVDA+q`, choose "Restart with add-ons disabled and debug logging enabled" and validate. Do not use another way to restart: `control+alt+n` or mouse would instead focus the task bar
* When NVDA has restarted, you should already be in the webpage of the browser, no need to press `tab`, `alt+tab` or whatever.
* Read the page. As expected due to #13704, NVDA uses UIA for Chrome. So math is not read with MathCAT (since MathCAT does not work with Chrome UIA; see #19852).
* Move the focus away and back (e.g. focusing the window menu with pressing `alt`, or with `alt+tab` to another application) to trigger a focus event so that "Use UIA with Microsoft Edge and other Chromium based browsers when available" option = "Only when necessary" be taken into account, i.e. Chrome should finally use IAccessible.
* Try to read math
* Press `NVDA+F1` to open the log, then `control+S` to save it to disk, e.g. in `nvda.log` file.
* Press `windows+R`, type `%tmp%` and press `enter` to go to temporary folder
* Look for `nvda-old.log` file
* Copy both `nvda.log` and `nvda-old.log` to your GitHub answer.
Thanks!

**Comment by GeorgeKerscher** [comment]
I have done as you asked. Below are links to the files. This replicates the issue. I also replicated in an XHTML that has MathML. I had the same result.
[NVDA log](https://www.dropbox.com/scl/fi/g9g9ekiyklsc63m6j52us/nvda.log?rlkey=547inivsr9dwognux707xtt1c&dl=0)
[Old NVDA log](https://www.dropbox.com/scl/fi/h62e0wqu1n6vggux7kawy/nvda-old.log?rlkey=jhedrxz3g0esdwdf1ic0ujjfv&dl=0)

**Comment by CyrilleB79** [comment]
@SaschaCowley or @RyanMcCleary have you been able to identify the cause of the issue? Yes, there is a workaround. But it's quite frustrating, when the issue occurs, to have to close all open Chrome windows / tabs to have math read again. Since I restart NVDA more often than a standard user, I may be facing this issue more often. But I wonder if restarting NVDA while in Chrome is the only case where this issue occurs though; I am not sure at all.

If you have had the opportunity to investigate and if the fix is not too heavy, it may be worth re-targeting it to 2026.1.

**Comment by CyrilleB79** [comment]
In addition to my previous message:
I have just realized that math cannot be read if a webpage was already opened in the browser before NVDA is started or restarted, no matter if the browser is actually in foreground or not when NVDA is (re)started. This means that this issue is quite likely to occur, depending on how people use NVDA:
* if they sometimes restart
* if their system opens a browser's page before NVDA is started.

Providing the first release including MathCAT with math reading failing so often will probably give a bad first impression of this new feature.

@SaschaCowley or @seanbudd given this, any chance to re-target this issue for 2026.1?

**Comment by RyanMcCleary** [comment]
@CyrilleB79 I have not yet been able to identify the cause of this issue. Which of the following do you think is more likely?
- Chrome is using UIA even after a focus event is triggered, when it should be using IA2, resulting in MathML not being picked up by MathCAT, or
- Chrome is correctly using IA2, but MathCAT is still not getting the MathML for some other reason

Could this issue and #19852 be originating from the same cause?

**Comment by CyrilleB79** [comment]
Given the `DEBUGWARNING` logged by `NVDAObjects.IAccessible.ia2Web.Math._get_mathMl`, I'd say that Chrome is using IA2 too. Or do you mean that Chrome could possibly use UIA while NVDA tries to use IA2? I do not know these interfaces on browser / app (provider) side to say if it is only possible.

**Comment by michaelDCurran** [comment]
For some reason iSimpleDOM may not be getting registered. I'm not too sure why. However, Chromium browsers now expose a 'math' IAccessible2 attribute on the math object, which contains all the raw mathML. Thus, we can check for this attribute first, and if it exists, pass that math content to mathCat, and not rely on ISimpleDOM at all.
